crates/services/consensus_module/poa/src/service.rs

```diff
@@ -158,6 +177,29 @@ where
     fn next_height(&self) -> BlockHeight {
         self.last_height + 1u32.into()
     }
+
+    fn next_time(&self, request_type: RequestType) -> anyhow::Result<Tai64> {
+        let now = Tai64::now();
+        match request_type {
+            RequestType::Manual => match self.trigger {
+                Trigger::Never | Trigger::Instant => self.next_time(RequestType::Trigger),
+                Trigger::Interval { block_time } => {
+                    increase_time(self.last_timestamp, block_time)
+                }
+                Trigger::Hybrid { min_block_time, .. } => {
+                    increase_time(self.last_timestamp, min_block_time)
+                }
+            },
+            RequestType::Trigger => {
+                if now > self.last_timestamp {
+                    Ok(now)
+                } else {
+                    let duration = self.last_block_created.elapsed();
```

Comment:
hmm should we still preserve the interval time period between blocks if manual mode is enabled? Otherwise if this gets triggered during a test it will likely start creating blocks at the same height for a while.